src/main/js/diff/Diff.tsx

```diff
@@ -64,99 +65,81 @@ const CommentWrapper = styled.div`
   }
 `;
 
-export type FileCommentState = {
-  comments: string[];
-  editor?: boolean;
-};
-
-export type FileCommentCollection = {
-  // path
-  [key: string]: FileCommentState;
-};
-export type LineCommentCollection = {
-  // hunkid
-  [key: string]: {
-    // changeid
-    [key: string]: {
-      location: Location;
-      comments: string[];
-      editor?: boolean;
-    };
-  };
-};
-
-export type DiffState = {
-  files: FileCommentCollection;
-  lines: LineCommentCollection;
-  comments: Comment[];
-  reviewedFiles: string[];
-};
-
 type Props = {
   repository: Repository;
   pullRequest: PullRequest;
+  comments?: Comments;
   diffUrl: string;
-  diffState: DiffState;
-  updateDiffState: (state: DiffState) => void;
   createLink?: string;
   fileContentFactory: FileContentFactory;
-};
-
-const openEditor = (
-  diffState: DiffState,
-  updateDiffState: (state: DiffState) => void,
-  editor: boolean,
-  location?: Location
-) => {
-  if (isInlineLocation(location)) {
-    const lineComments = diffState.lines;
-    const hunkId = createHunkIdFromLocation(location!);
-    const changeId = createChangeIdFromLocation(location!);
-    const hunkComments = lineComments[hunkId] || {};
-    const changeComments = hunkComments[changeId] || {
-      location,
-      comments: []
-    };
-
-    changeComments.editor = editor;
-    hunkComments[changeId] = changeComments;
-    lineComments[hunkId] = hunkComments;
-    updateDiffState({ ...diffState, lines: lineComments });
-  } else {
-    const fileComments = diffState.files;
-    const file = location!.file;
-    const fileState = fileComments[file] || {
-      comments: []
-    };
-    fileState.editor = editor;
-    fileComments[file] = fileState;
-    updateDiffState({ ...diffState, files: fileComments });
-  }
+  reviewedFiles: string[];
 };
 
 const Diff: FC<Props> = ({
   repository,
   pullRequest,
+  comments,
   diffUrl,
-  diffState,
-  updateDiffState,
   createLink,
-  fileContentFactory
+  fileContentFactory,
+  reviewedFiles
 }) => {
   const [t] = useTranslation("plugins");
   const { actions, isCollapsed } = useDiffCollapseState(pullRequest);
   const [collapsed, setCollapsed] = useState(false);
+  const [openEditors, setOpenEditors] = useState<{ [hunkId: string]: string[] }>({});
+
+  const openEditor = (editor: boolean, location: Location) => {
```

Comment:
I would prefer two methods here to make it clear: `openEditor` and `closeEditor`. Besides the first `if`, the logic is totally disjunct.

Reply:
You can even write two dedicated methods for `openEditor` without the need the first if, because in case of "open" you know whether it's inline or file related. So you could have three methods in sum: `openInlineEditor`, `openFileEditor`, and `closeEditor`.

Reply:
You are right. It's way more readable. :+1: